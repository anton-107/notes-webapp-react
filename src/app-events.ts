import React from "react";
import EventEmitter from "eventemitter3";

export type ApplicationEventType = "notebook.deleted" | "notebook.added";

export class ApplicationEventEmitter {
  private eventEmitter = new EventEmitter();

  public addListener(eventType: ApplicationEventType, handler: () => void) {
    this.eventEmitter.on(eventType, handler);
  }
  public removeListener(eventType: ApplicationEventType, handler: () => void) {
    this.eventEmitter.off(eventType, handler);
  }
  public emitEvent(eventType: ApplicationEventType) {
    this.eventEmitter.emit(eventType);
  }
}
export const ApplicationEvents = React.createContext(
  new ApplicationEventEmitter()
);
