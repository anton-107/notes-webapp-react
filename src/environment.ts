/// <reference types="vite/client" />

interface ImportMetaEnv {
  readonly VITE_API_ROOT: string;
}

// eslint-disable-next-line @typescript-eslint/no-unused-vars
interface ImportMeta {
  readonly env: ImportMetaEnv;
}

const API_ROOT: string | undefined = import.meta.env.VITE_API_ROOT;

export { API_ROOT };
