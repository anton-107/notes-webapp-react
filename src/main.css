html,
body {
  padding: 0;
  margin: 0;
  overflow: hidden;
  font-family: -apple-system, BlinkMacSystemFont, Roboto, "Helvetica Neue",
    Helvetica, Arial, sans-serif;
}
.layout {
  display: flex;
}
.vertical-menu {
  background-color: #000;
  color: #fff;
  width: 250px;
  height: 100vh;
}
.content {
  flex: 1;
  height: 100vh;
}
.menu-block {
  padding-left: 30px;
  font-size: 1em;
}
.menu-links {
  list-style-type: none;
  margin: 0;
  padding: 0;
}
.menu-links li {
  margin: 0;
  padding-left: 30px;
}
.menu-links li a {
  display: block;
  padding-top: 10px;
  padding-bottom: 10px;
  color: #fff;
}
.menu-links li a:hover {
  background-color: rgba(255, 255, 255, 0.08);
}
.menu-header {
  font-size: 1.2em;
  line-height: 2.4em;
}
.content-block {
  padding: 60px;
}
