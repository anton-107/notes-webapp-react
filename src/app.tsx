import * as React from "react";
import "./main.css";

export function App(props: { name: string }): React.ReactElement {
  return (
    <div id="app-root">
      <div className="layout">
        <div className="vertical-menu">
          <h1 className="menu-block menu-header">Notes app</h1>
          <ul className="menu-links">
            <li>
              <a href="#">Notebooks</a>
            </li>
            <li>
              <a href="#">People</a>
            </li>
          </ul>
        </div>
        <div className="content">
          <div className="content-block">Hello {props.name}</div>
        </div>
      </div>
    </div>
  );
}
