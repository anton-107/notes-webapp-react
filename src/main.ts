/* istanbul ignore file */
import * as React from 'react';
import {createRoot} from 'react-dom/client';
import { App } from './app';

function run() {
  const staticPage = document.getElementById('static-page');
  if (staticPage) {
    staticPage.innerHTML = '';
  }
  const domContainer = document.querySelector('#notes-webapp-react');
  const root = createRoot(domContainer);
  root.render(React.createElement(App, {name: 'anonymous'}));
}

run();