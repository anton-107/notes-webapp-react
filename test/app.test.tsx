import * as React from "react";
import { create } from "react-test-renderer";
import { App } from "../src/app";

describe("App", () => {
  it("should match a snapshot", () => {
    const component = create(<App name="anonymous" />);
    const tree = component.toTree();
    expect(tree).toMatchSnapshot();
  });
});
