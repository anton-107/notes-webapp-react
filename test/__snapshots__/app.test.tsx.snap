// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match a snapshot 1`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {
    "name": "anonymous",
  },
  "rendered": Object {
    "instance": null,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="layout"
      >
        <div
          className="vertical-menu"
        >
          <h1
            className="menu-block menu-header"
          >
            Notes app
          </h1>
          <ul
            className="menu-links"
          >
            <li>
              <a
                href="#"
              >
                Notebooks
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                People
              </a>
            </li>
          </ul>
        </div>
        <div
          className="content"
        >
          <div
            className="content-block"
          >
            Hello 
            anonymous
          </div>
        </div>
      </div>,
      "id": "app-root",
    },
    "rendered": Array [
      Object {
        "instance": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="vertical-menu"
            >
              <h1
                className="menu-block menu-header"
              >
                Notes app
              </h1>
              <ul
                className="menu-links"
              >
                <li>
                  <a
                    href="#"
                  >
                    Notebooks
                  </a>
                </li>
                <li>
                  <a
                    href="#"
                  >
                    People
                  </a>
                </li>
              </ul>
            </div>,
            <div
              className="content"
            >
              <div
                className="content-block"
              >
                Hello 
                anonymous
              </div>
            </div>,
          ],
          "className": "layout",
        },
        "rendered": Array [
          Object {
            "instance": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1
                  className="menu-block menu-header"
                >
                  Notes app
                </h1>,
                <ul
                  className="menu-links"
                >
                  <li>
                    <a
                      href="#"
                    >
                      Notebooks
                    </a>
                  </li>
                  <li>
                    <a
                      href="#"
                    >
                      People
                    </a>
                  </li>
                </ul>,
              ],
              "className": "vertical-menu",
            },
            "rendered": Array [
              Object {
                "instance": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Notes app",
                  "className": "menu-block menu-header",
                },
                "rendered": Array [
                  "Notes app",
                ],
                "type": "h1",
              },
              Object {
                "instance": null,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <li>
                      <a
                        href="#"
                      >
                        Notebooks
                      </a>
                    </li>,
                    <li>
                      <a
                        href="#"
                      >
                        People
                      </a>
                    </li>,
                  ],
                  "className": "menu-links",
                },
                "rendered": Array [
                  Object {
                    "instance": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": <a
                        href="#"
                      >
                        Notebooks
                      </a>,
                    },
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Notebooks",
                          "href": "#",
                        },
                        "rendered": Array [
                          "Notebooks",
                        ],
                        "type": "a",
                      },
                    ],
                    "type": "li",
                  },
                  Object {
                    "instance": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": <a
                        href="#"
                      >
                        People
                      </a>,
                    },
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "nodeType": "host",
                        "props": Object {
                          "children": "People",
                          "href": "#",
                        },
                        "rendered": Array [
                          "People",
                        ],
                        "type": "a",
                      },
                    ],
                    "type": "li",
                  },
                ],
                "type": "ul",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="content-block"
              >
                Hello 
                anonymous
              </div>,
              "className": "content",
            },
            "rendered": Array [
              Object {
                "instance": null,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Hello ",
                    "anonymous",
                  ],
                  "className": "content-block",
                },
                "rendered": Array [
                  "Hello ",
                  "anonymous",
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  "type": [Function],
}
`;
